common/src/edgeai_demo.cpp

```diff
@@ -0,0 +1,1191 @@
+                                               (vx_reference*) &input_o,
+                                               1,
+                                               &num_refs);
+            }
+        }
+
+        for(uint32_t i = 0; i < m_imgMosaicObjs.size(); i++)
+        {
+            if( (i == m_dispMosaicIdx) && (m_displayObj != NULL) )
+            {
+                continue;
+            }
+
+            vxGraphParameterDequeueDoneRef(m_ovxGraph->graph,
+                                           m_imgMosaicObjs[i]->imgMosaicObj.output_graph_parameter_index,
+                                           (vx_reference*) &output_o,
+                                           1,
+                                           &num_refs);
+            vxGraphParameterDequeueDoneRef(m_ovxGraph->graph,
+                                           m_imgMosaicObjs[i]->imgMosaicObj.background_graph_parameter_index,
+                                           (vx_reference*) &background_o,
+                                           1,
+                                           &num_refs);
+        }
+
+        /*  Write Output to directory or file if needed. */
+        for(uint32_t i = 0; i < m_imgMosaicObjs.size(); i++)
+        {
+            string sink = "";
+            string sink_type = "";
+            for (auto &[name,flow] : m_config.m_flowMap)
+            {
+                auto const &outputIds = flow->m_outputIds;
+                for(uint32_t j=0; j < outputIds.size(); j++)
+                {
+                    OutputInfo  *output = m_config.m_outputMap[outputIds[j]];
+                    if ((uint32_t)output->m_instId == i)
+                    {
+                        sink = output->m_sink;
+                        sink_type = output->m_sinkType;
+                        break;
+                    }
+                }
+                if (sink != "")
+                {
+                    break;
+                }
+            }
+
+            if(sink_type == "image")
+            {
+                writeImage(const_cast<char*>(sink.c_str()),
+                           m_imgMosaicObjs[i]->imgMosaicObj.output_image[0]);
+            }
+            else if(sink_type == "images_directory")
+            {
+                string mosaicId = "mosaic" + to_string(i);
+                string output_filename = sink +
+                                         "/output_image_" +
+                                         mosaicId +
+                                         "_" +
+                                         to_string(cnt) +
+                                         ".nv12";
+                writeImage(const_cast<char*>(output_filename.c_str()),
+                                             m_imgMosaicObjs[i]->imgMosaicObj.output_image[0]);
+            }
+        }
+        cnt++;
+
+        /* HACK: If no camera input then sleep of 1 second to maintain FPS for image input */
```

Comment:
Is this really a hack ... it sounds like a solution for a "feature".  Is there a better way to do this, and that is why it is a "hack", if so, what is the "right" way to do it?

Reply:
Agreed. For now this looks best possible solution. Updated the PR